assembl/models/discussion.py

```diff
@@ -31,12 +31,51 @@
 from .langstrings import LangString
 from .preferences import Preferences
 from assembl.lib.caching import create_analytics_region
-
+from assembl.graphql import docstrings as docs
 resolver = DottedNameResolver(__package__)
 log = logging.getLogger()
 visit_analytics_region = create_analytics_region()
 
 
+class OldSlug(DiscussionBoundBase, NamedClassMixin):
+    """
+    The old slug of a debate. Used in case the administrator changes the slug of a debate.
+    """
+    __tablename__ = "old_slug"
+    id = Column(Integer, primary_key=True)
+    discussion_id = Column(Integer, ForeignKey(
+        'discussion.id',
+        ondelete='CASCADE',
+        onupdate='CASCADE'),
+        nullable=False,
+        index=True)
+    discussion = relationship(
+        "Discussion",
+        backref=backref(
+            'old_slugs',
+            cascade="all, delete-orphan"),
+    )
+    slug = Column(CoerceUnicode, nullable=False, unique=True, index=True, doc=docs.OldSlug.slug)
+    redirection_slug = Column(CoerceUnicode, nullable=False, unique=True, index=True, doc=docs.OldSlug.redirection_slug)
+
+    @classmethod
+    def get_discussion_id(self):
+        return self.discussion_id or self.discussion.id
+
+    @classmethod
+    def get_discussion_conditions(cls, discussion_id, alias_maker=None):
+        return (cls.discussion_id == discussion_id,)
+
+    @classmethod
+    def push_old_slug(cls, discussion_id, slug):
+        all_slugs = cls.default_db.query(cls.slug).all()
+        discussion = cls.default_db.query(Discussion).get(discussion_id)
```

Comment:
Instead of getting all slugs, you just get only the get the slug you're interested in. If it's None, create it.

Reply:
This class method is only used in a test. So remove it completly or use it in the UpdateDiscussionPreferences mutation.